include/cell.tcc

```diff
@@ -856,6 +856,45 @@ Eigen::VectorXd mpm::Cell<Tdim>::compute_strain_rate(
   return strain_rate;
 }
 
+//! Compute strain rate for reduced integration at the centroid of cell
+template <unsigned Tdim>
+Eigen::VectorXd mpm::Cell<Tdim>::compute_strain_rate_centroid(unsigned phase) {
+  // Define strain rate
+  Eigen::VectorXd strain_rate;
+
+  switch (Tdim) {
```

Comment:
Instead of these switches, you could just do `bmatrix.at(0).rows()` to get the size of your strain_rate_centrodi.